api/config/v1alpha1/shared_types.go

```diff
@@ -56,6 +58,11 @@ type KubernetesDeploymentSpec struct {
 	// +optional
 	Resources *corev1.ResourceRequirements `json:"resources,omitempty"`
 
+	// Image specifies the image used by the container.
```

Comment:
based on my comment, can we add some logic that runs `envoy --version` and ensures the version is same ?

Reply:
this can be done in a follow up PR